docs/tutorial/statistics.md

```diff
@@ -0,0 +1,124 @@
+# Statistics
+
+Concrete analyzes all compiled circuits and calculates some statistics. These statistics can be used to find bottlenecks and compare circuits. Statistics are calculated in terms of basic operations. There are 6 basic operations in Concrete:
+
+- **clear addition:** x + y where x is encrypted and y is clear
+- **encrypted addition:** x + y where both x and y are encrypted
+- **clear multiplication:** x * y where x is encrypted and y is clear
+- **encrypted negation:** -x where x is encrypted
+- **key switch:** building block for table lookups
+- **packing key switch:** building block for table lookups
+- **programmable bootstrap:** building block for table lookups
+
+You can print all statistics using `show_statistics` configuration option:
+
+```python
+from concrete import fhe
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return (x**2) + (2*x) + 4
+
+inputset = range(2**2)
+circuit = f.compile(inputset, show_statistics=True)
+```
+
+This code will print:
+
+```
+Statistics
+--------------------------------------------------------------------------------
+size_of_secret_keys: 22648
+size_of_bootstrap_keys: 51274176
+size_of_keyswitch_keys: 64092720
+size_of_inputs: 16392
+size_of_outputs: 16392
+p_error: 9.627450598589458e-06
+global_p_error: 9.627450598589458e-06
+complexity: 99198195.0
+programmable_bootstrap_count: 1
+programmable_bootstrap_count_per_parameter: {
+    BootstrapKeyParam(polynomial_size=2048, glwe_dimension=1, input_lwe_dimension=781, level=1, base_log=23, variance=9.940977002694397e-32): 1
+}
+key_switch_count: 1
+key_switch_count_per_parameter: {
+    KeyswitchKeyParam(level=5, base_log=3, variance=1.939836732335308e-11): 1
+}
+packing_key_switch_count: 0
+clear_addition_count: 1
+clear_addition_count_per_parameter: {
+    LweSecretKeyParam(dimension=2048): 1
+}
+encrypted_addition_count: 1
+encrypted_addition_count_per_parameter: {
+    LweSecretKeyParam(dimension=2048): 1
+}
+clear_multiplication_count: 1
+clear_multiplication_count_per_parameter: {
+    LweSecretKeyParam(dimension=2048): 1
+}
+encrypted_negation_count: 0
+--------------------------------------------------------------------------------
+```
+
+{% hint style="info" %}
+Each of these properties can be directly accessed on the circuit (e.g., `circuit.programmable_bootstrap_count`).
+{% endhint %}
+
```

Comment:
is it possible to say few words about the parameters which make PBS slow or fast? like, saying "typically, a BootstrapKeyParam(polynomial_size=2048) would go much slower than a BootstrapKeyParam(polynomial_size=1024)". And maybe relate to the bitwidth of inputs of PBS and the norm2. But I know it's not easy to explain, do your best

Reply:
I don't think it's a good idea. Cost model is complex, I think the better thing to do would be to add complexity to the object.

Reply:
ok but it's not obvious for users to understand eg that a PBS(2048) is not to be compared with a PBS(512)

Reply:
That's why I think we should add complexity.
CC @rudy-6-4

Reply:
Is complexity a new dev to do, or is it just about the documentation?

Reply:
(just to mention it, there is a sentence here about PBS cost https://docs.zama.ai/concrete/getting-started/fhe_basics#pbs-management)

Reply:
It's a new dev to do, basically we'll convert:
```python
BootstrapKeyParam(polynomial_size=2048, glwe_dimension=1, input_lwe_dimension=781, level=1, base_log=23, variance=9.940977002694397e-32)
```
to
```python
BootstrapKeyParam(complexity=xxx, polynomial_size=2048, glwe_dimension=1, input_lwe_dimension=781, level=1, base_log=23, variance=9.940977002694397e-32)
```
and order by complexity.

Reply:
ok so could you make an issue if not already done, and discuss with @rudy-6-4 ? but it's out of scope of this PR / too late for this delivery, please

Reply:
https://github.com/zama-ai/concrete-internal/issues/481 :+1: 

It won't be delivered in this PR.